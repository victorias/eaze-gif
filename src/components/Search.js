// @flow

import React, { PureComponent } from 'react';
import styled from 'styled-components';

type Props = {
  search: (query: string) => void,
  getTrending: () => void,
};

type State = {
  query: string,
};

const Input = styled.input`
  display: flex;
  outline: none;
  border-top: 0;
  border-left: 0;
  border-right: 0;
  padding-bottom: 14px;
  color: #1b395d;
  -webkit-text-fill-color: #1b395d;
  letter-spacing: 0.5px;
  background: transparent;
  border-radius: 0px;
  font-size: 40px;
  border-bottom: 1px solid rgba(27, 57, 93, 0.3);
  font-family: 'Roboto';
`;

class Search extends PureComponent<Props, State> {
  state = { query: 'trending' };

  render() {
    const { props, state } = this;
    return (
      <Input
        type="text"
        value={state.query}
        onChange={event => this.setState({ query: event.target.value })}
        onKeyPress={e => {
          if (e.key === 'Enter') {
            if (state.query === 'trending') {
              props.getTrending();
            } else {
              props.search(state.query);
            }
          }
        }}
      />
    );
  }
}

export default Search;
