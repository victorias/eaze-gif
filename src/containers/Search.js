// @flow

import { connect } from 'react-redux';
import Search from '../components/Search';

import * as actions from '../redux/gifs/actions';

import type { Dispatch } from '../redux/gifs/actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  search: query => dispatch(actions.search(query)),
  getTrending: () => dispatch(actions.getTrending()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Search);
