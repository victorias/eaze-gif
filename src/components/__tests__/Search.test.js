// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Search from '../Search';

describe('Search', () => {
  it('renders', () => {
    const tree = renderer.create(
      <Search search={q => undefined} getTrending={() => undefined} />
    );
    expect(tree).toMatchSnapshot();
  });
});
