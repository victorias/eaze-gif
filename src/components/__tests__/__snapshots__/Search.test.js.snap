// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders 1`] = `
<input
  className="sc-bdVaJa fanUGB"
  onChange={[Function]}
  onKeyPress={[Function]}
  type="text"
  value="trending"
/>
`;
